install/kubernetes/helm/istio/templates/configmap.yaml

```diff
@@ -72,7 +72,7 @@ data:
     {{- if .Values.ingress.enabled }}
     # This is the k8s ingress service name, update if you used a different name
     ingressService: istio-{{ .Values.global.k8sIngressSelector }}
-    {{- else }}
+    {{- else if and .Values.gateways.enabled }}
```

Comment:
is this how the helm thing works? shouldn't this be `else if .Values.gateways.enabled` ?

Also, why does this even need an else? IIRC, the ingressService is required in Pilot for the status update in k8s to specify the public IP address of the ingress controller, so that people can do `kubectl get ing`. Incorrectly marking it as istio-ingressgateway doesn't sound right, when the user has disabled the ingress controller completely.

Reply:
sorry I was trying to co-guard with .Values.gateways.istio-ingress.enabled but that has its own parsing prob due to non compliant value for helm.  

I think the else is useful when we auto convert the ingress to ingress gateway yaml so people at least know to refer to the ingress gateway ip when looking it up via kubectl get ing.  @costinm your input pls?